controllers/addon_controller.go

```diff
@@ -173,71 +173,77 @@ func (r *AddonReconciler) execAddon(ctx context.Context, req reconcile.Request,
 	return ret, procErr
 }
 
-// SetupWithManager is called to setup manager and watchers
-func (r *AddonReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	log := r.Log
-	managedNS := "addon-manager-system"
-
-	nsInformers := dynamicinformer.NewFilteredDynamicSharedInformerFactory(r.dynClient, time.Minute*30, managedNS, nil)
-	wfInf := nsInformers.ForResource(common.WorkflowGVR())
-	bldr := ctrl.NewControllerManagedBy(mgr).
-		For(&addonmgrv1alpha1.Addon{}).
-		// Watch workflows created by addon only in addon-manager-system namespace
-		Watches(&source.Informer{Informer: wfInf.Informer().(cache.Informer)}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &addonmgrv1alpha1.Addon{},
-		})
-
-	generatedInformers = informers.NewSharedInformerFactory(r.generatedClient, time.Minute*30)
-
-	err := mgr.Add(manager.RunnableFunc(func(s <-chan struct{}) error {
-		generatedInformers.Start(s)
-		generatedInformers.WaitForCacheSync(s)
-		nsInformers.Start(s)
-		nsInformers.WaitForCacheSync(s)
-		<-s
-		return nil
-	}))
+func New(mgr manager.Manager) (controller.Controller, error) {
+	r := &AddonReconciler{
+		Client:          mgr.GetClient(),
+		Log:             ctrl.Log.WithName(controllerName),
+		Scheme:          mgr.GetScheme(),
+		versionCache:    addon.NewAddonVersionCacheClient(),
+		dynClient:       dynamic.NewForConfigOrDie(mgr.GetConfig()),
+		generatedClient: kubernetes.NewForConfigOrDie(mgr.GetConfig()),
+		recorder:        mgr.GetEventRecorderFor("addons"),
+		statusWGMap:     map[string]*sync.WaitGroup{},
+	}
+
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
 	if err != nil {
-		log.Error(err, "Error adding InformerFactory to the Manager")
-		return err
+		return nil, err
+	}
+
+	// Watch workflows created by addon only in addon-manager-system namespace
+	if err := c.Watch(&source.Kind{Type: &wfv1.Workflow{}}, &handler.EnqueueRequestForOwner{
```

Comment:
This cannot work, we must only watch Workflows in the controller namespace. Before we limited to the controller ns here:
```
nsInformers := dynamicinformer.NewFilteredDynamicSharedInformerFactory(r.dynClient, time.Minute*30, managedNS, nil)
	wfInf := nsInformers.ForResource(common.WorkflowGVR())
	bldr := ctrl.NewControllerManagedBy(mgr).
		For(&addonmgrv1alpha1.Addon{}).
		// Watch workflows created by addon only in addon-manager-system namespace
		Watches(&source.Informer{Informer: wfInf.Informer().(cache.Informer)}, &handler.EnqueueRequestForOwner{
			IsController: true,
			OwnerType:    &addonmgrv1alpha1.Addon{},
		})
```

Reply:
when checking with resources, we filter using labelselector which pass namespace and name which is aligned specifically to the addon resource(s), which is implemented within the objects module.
thus, we do not need informers anymore

Reply:
We have previously fixed issue for memory consumption due to Watch (using the builder) see #73, which may be specifically an issue w/ CacheInformers but we also don't want to watch Workflows across all cluster, the design of Addon Manager at least for now is to run workflows only in `addon-manager-system` namespace so setting up a Watch for cluster is not efficent in the least and would be a regression in the worst case.

Reply:
Only watch addon-manager-system namespace workflow
https://github.com/keikoproj/addon-manager/pull/96/commits/63d89c2dd79f8f770b3ae5a95f39f37e185d13af#diff-2df6a56a33260adb76f704ff7c6737be9f43c13fe523521ee76390ff2a9ca4b8R242